src/LeagueToolkit/Meta/MetaEmbedded.cs

```diff
@@ -9,17 +9,17 @@ public T Value
             get => this._value;
             set
             {
-                ArgumentNullException.ThrowIfNull(value, nameof(value));
-                this._value = value;
+                if (value is null) throw new ArgumentNullException(nameof(value));
```

Comment:
Can change to use `Guard` API

Reply:
At this point I could also change all occurences of the pattern `if (x is null) throw new ArgumentNullException()` to use the Guard syntax if that's fine? Not sure why I even included this specific one in the first place.